File: week6/pset6/mario.py
import cs50

def main():
    height = get_height()
    print_pyramid(height)

def get_height():
    n = 0
    while(True):
        n = input("Height: ")
        if n.isnumeric():
            if 0 < int(n) < 9:
                break
    return int(n)

def print_pyramid(height):
    for count in range(height):
        space = height - (count + 1)
        brick = height - space
        for i in range(space):
            print(" ", end="")
        for i in range(brick):
            print("#", end="")
        print("  ", end="")
        for i in range(brick):
            print("#", end="")
        print()




main()

File: week1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt for height
    int height;
    do
    {
        height = get_int("Pyramid height: ");
    }
    while (height < 1 || height > 8);

//Initialize stats
    int c = 0;
    int h = height;

//Opener
    while (c < height)
    {
        int s = h - 1;
        int b = c+1;

//Space feature 1
        while (s>0)
        {
            printf(" ");
            s--;
        }
//Brick feature 1
        while (b>0)
        {
            printf("#");
            b--;
        }
//Midspace!
        printf("  ");

//Brick feature 2
        while (b < ( c + 1 ))
        {
            printf("#");
            b++;
        }

//Closure (don't we all need some?)
    c++;
    h--;
    printf("\n");
    }
}